server/etcdserver/api/v3rpc/maintenance.go

```diff
@@ -203,12 +203,16 @@ func (ms *maintenanceServer) Hash(ctx context.Context, r *pb.HashRequest) (*pb.H
 }
 
 func (ms *maintenanceServer) HashKV(ctx context.Context, r *pb.HashKVRequest) (*pb.HashKVResponse, error) {
-	h, rev, compactRev, err := ms.kg.KV().HashByRev(r.Revision)
+	rev := r.Revision
+	if rev == 0 {
+		rev = ms.kg.KV().Rev()
+	}
```

Comment:
Goal of this change is to simplify `KV.HashByRev` method, by removing this special case that could cause become a source of bugs in future. From KV point of view, it doesn't makes sense that `0` revision has special meaning of latest. This is more of an API than storage behavior. 

I think lower level code should follow rule "require more, promise less". It's ok to a lot of promises at the API level, however it can be disruptive for lower level code. With proposed changes `HashByRev` will be called for other use cases than just API response as so we should avoid special cases as much as possible. 

This is tradeoff on clarity vs performace. For me here clarity is more important, as performance impact seems not very significant. This change would require additional `concurentReadTx` lock grap on every HashByRev request. This request is not called very often as so I think it is acceptable. 

Please let me know if it makes sense or I missed anything.

Reply:
This change was reimplemented so the comment no longer applies.